src/app/icd/ICDClientInfoManagement.cpp

```diff
@@ -0,0 +1,389 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines a basic implementation of ICDClientPersistentStorage that
+ *      persists clientInfo in a flat list in TLV.
+ */
+
+#include <app/icd/ICDClientInfoManagement.h>
+
+#include <lib/core/Global.h>
+#include <lib/support/Base64.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace app {
+
+Global<ICDClientInfoManagement> sICDClientInfoManagement;
+
+ICDClientInfoManagement * ICDClientInfoManagement::GetInstance()
+{
+    return &sICDClientInfoManagement.get();
+}
+
+ICDClientInfoManagement::ICDClientInfoIteratorImpl::ICDClientInfoIteratorImpl(ICDClientInfoManagement & aStorage) :
```

Comment:
What sort of clients are we targeting here?

Because if the idea is "capable clients", this seems like the wrong general API surface....

Reply:
yes, capable clients, ICDClientInfoIteratorImpl would be exposed iterator for application that traverse all keys when check-in message comes, for example.